core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -160,26 +160,6 @@ public IndexRequest() {
     }
 
     /**
-     * Copy constructor that creates a new index request that is a copy of the one provided as an argument.
-     * The new request will inherit though headers and context from the original request that caused it.
-     */
-    public IndexRequest(IndexRequest indexRequest) {
```

Comment:
It's annoying how these things are error-prone, like you add a new instance member and you forget to add it to the copy constructor (just happened with pipeline, which triggered this whole PR). Would be nice to find another way :)